### ISSUE TITLE
Question: Blastula to Python

### ISSUE THREAD

**Issue by BlueyCoder** [rstudio/blastula#325]
Hi,

I am not sure if this is the right forum for this but if not please close it.
`Blastula` has been a godsend for me in my job, allowing me to automate a lot of reports and ML workflows from my laptop.
I was wondering, given Posit's wider scope into the Python ecosystem, are there plans to have a Python variant of the package

Thank you very much for developing the package

All the best

**Comment by rich-iannone** [comment]
@BlueyCoder Sorry this took a very long time to answer. I've been doing much more Python work lately so a Python port seems increasingly likely. Question for you: what currently exists in the Python space that's comparable to blastula? Or do people just use the built-in emailing functionality in Python?

cc @jrycw, this could be an opportunity for a new Python package. Perhaps it wouldn't be as much work in Python if RFC-2822 support is already there (see https://github.com/rstudio/blastula/blob/081853a6a30367669867ef8f9aa9ca72c6c5b31f/R/mime.R#L39 for what we did for this conversion in R).

**Comment by BlueyCoder** [comment]
No worries about the length of time responding.
I am not sure how the rest of the python community does it but i cheat a little bit.
I develop the logic of the report or app in Python and then run it in R.
I would write the markdown file import it in and then insert the variables and graphs and send it out.
I found the normal Python way too verbose to maintain.

```
# RUN PYTHON FILE ---------------------------------------------------------
# Use the specified Python environment
use_python('C:/Users/foo/Miniconda3/envs/spyder-env/python.exe')
source_python("C:/Users/foo/1-main.py")

print('PYTHON SCRIPT RUN')

# CREATE SOME CHARTS IN R

# CREATE EMAIL CONTENT AND SEND -------------------------------------
# Create the variables for the email
charts <- add_ggplot(plot_object = ggcharts, width = 16, height = 8)
email_content <- read_file(glue('{wkdir}/email_template.md')) 
email_content <- glue(email_content)
email <- blastula::compose_email(body = md(email_content))

email %>%
  smtp_send() # etc etc

```

**Comment by rich-iannone** [comment]
This is great to see, thanks! Seeing this makes me more convinced that a Python blastula (probably won't use the same name) is a good idea in 2025.

**Comment by jrycw** [comment]
This is a great idea. `Python blastula` could be offered as an optional package for sending reports with `Pointblank`, especially for users who find the built-in email feature cumbersome to use.

**Comment by rich-iannone** [comment]
@jrycw agree! And it would fit in nicely with a group of notifier actions like this one: https://posit-dev.github.io/pointblank/reference/send_slack_notification.html

**Comment by BlueyCoder** [comment]
I am happy to test it when you get the time to develop it.
